analysis_driver/pipelines/human_variant_calling_gatk4.py

```diff
@@ -139,74 +133,78 @@ def _run(self):
-
-    final_stages = [contam, blast, geno_val, gender_val, vcfstats, verify_bam_id, samtools_depth, samtools_stat,
-                    gather_gcvf]
-
-    output = stage(common.SampleDataOutput, previous_stages=final_stages, output_fileset='gatk4_human_var_calling')
-    review = stage(common.SampleReview, previous_stages=[output])
-    cleanup = stage(common.Cleanup, previous_stages=[review])
-
-    return cleanup
+class HumanVarCallingGATK4(Pipeline):
+    toolset_type = 'gatk4_sample_processing'
+    _name = 'human_variant_calling_gatk4'
+
+    def build(self):
+        """Build the variant calling pipeline (for human)."""
+
+        # merge fastq to do contamination check
+        merge_fastqs = self.stage(common.MergeFastqs)
+        contam = self.stage(qc.FastqScreen, previous_stages=[merge_fastqs], fq_pattern=merge_fastqs.fq_pattern)
+        blast = self.stage(qc.Blast, previous_stages=[merge_fastqs], fastq_file=merge_fastqs.fq_pattern.replace('?', '1'))
+        geno_val = self.stage(qc.GenotypeValidation, fq_pattern=merge_fastqs.fq_pattern, previous_stages=[merge_fastqs])
+
+        # create fastq index then align and recalibrate via scatter-gather strategy
+        fastq_index = self.stage(FastqIndex)
+        split_bwa = self.stage(SplitBWA, previous_stages=[fastq_index])
+        merge_bam_dup = self.stage(MergeBamAndDup, previous_stages=[split_bwa])
+        base_recal = self.stage(ScatterBaseRecalibrator, previous_stages=[merge_bam_dup])
+        gather_bqsr = self.stage(GatherBQSRReport, previous_stages=[base_recal])
+        apply_bqsr = self.stage(ScatterApplyBQSR, previous_stages=[gather_bqsr])
+        merge_bam = self.stage(GatherRecalBam, previous_stages=[apply_bqsr])
+
+        # bam file QC
+        verify_bam_id = self.stage(qc.VerifyBamID, bam_file=merge_bam.recal_bam, previous_stages=[merge_bam])
+        samtools_stat = self.stage(common.SamtoolsStats, bam_file=merge_bam.recal_bam, previous_stages=[merge_bam])
+        samtools_depth = self.stage(qc.SamtoolsDepth, bam_file=merge_bam.recal_bam, previous_stages=[merge_bam])
+
+        # variants call via scatter-gather strategy
+        haplotype_caller = self.stage(SplitHaplotypeCallerVC, bam_file=merge_bam.recal_bam, previous_stages=[merge_bam])
+        gather_gcvf = self.stage(GatherGVCF, previous_stages=[haplotype_caller])
+        genotype_gcvf = self.stage(SplitGenotypeGVCFs, previous_stages=[haplotype_caller])
+        gather_vcf = self.stage(GatherVCFVC, previous_stages=[genotype_gcvf])
+
+        variant_file = gather_vcf.genotyped_vcf
+        steps_required = [gather_vcf]
+        if 'snpEff' in self.dataset.genome_dict:
+            # variant annotation
+            annotate_vcf = self.stage(VariantAnnotation, previous_stages=[gather_vcf])
+            variant_file = annotate_vcf.snps_effects_output_vcf
+            steps_required = [annotate_vcf]
+
+        # VQSR is disabled for now until we can make it perform better than hard filtering.
```

Comment:
I'd prefer it if this was traceable with a GitHub issue. Obviously given the circumstances...

Reply:
That one is actually on me. I'll put an issue up for it.